### PR TITLE
feat(FloatingActionButton): Create FloatingActionButton

### LINKED ISSUE #18: Create FloatingActionButton Component
https://material.io/guidelines/components/buttons-floating-action-button.html

### PR DESCRIPTION
**Pull request by ashwoodall**
Create FloatingActionButton. Resolves #18 

Demo: https://docs-kmzwbqrntp.now.sh/

### CODE REVIEW

**Comment by ceciliamvrie** [comment]
Just wanted to point out that whenever you hover over the icon the background is still square! 
I guess that could be for another pr later on, but so far its a great first implementation!

**Comment by ashwoodall** [comment]
Weird! I wasn't seeing that on my machine, but I tried updating ```overflow: hidden;``` on hover. LMK if that fixes that square.

**Comment by AriLFrankel** [comment]
@ashwoodall Sorry I left that question in a pending state when I added that label.

File: src/components/FloatingActionButton.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  width: 40px;
+`;
+
+const FloatingActionButton = styled.button.attrs({
+  'data-smc': 'FloatingActionButton',
+})`
+  color: #fff;
+  width: 56px;
+  height: 56px;
+  padding: 0;
```
[the remark this answers is not in the document]
Comment (ashwoodall) [reply src/components/FloatingActionButton.js]: It helps center the icon in the middle. Without it the ```padding: 0``` it weirdly aligns the icons slightly more top left in the circular base.

[the remark this answers is not in the document]
Comment (AriLFrankel) [reply src/components/FloatingActionButton.js]: got ya, thanks for the clarification 👍

**Pull request merged**